internal/webhook/datasciencecluster/integration_test.go

```diff
@@ -0,0 +1,142 @@
+func TestDataScienceCluster_Integration(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		name  string
+		setup func(ns string) []client.Object
+		test  func(g Gomega, ctx context.Context, k8sClient client.Client, ns string)
+	}{
+		{
+			name: "Allows creation if none exist",
+			setup: func(ns string) []client.Object {
+				return nil
+			},
+			test: func(g Gomega, ctx context.Context, k8sClient client.Client, ns string) {
+				dsc := shared.NewDSC("dsc-one", ns)
+				g.Expect(k8sClient.Create(ctx, dsc)).To(Succeed(), "should allow creation of a DataScienceCluster when none exist")
+			},
+		},
+		{
+			name: "Denies creation if one already exists",
+			setup: func(ns string) []client.Object {
+				return []client.Object{
+					shared.NewDSC("existing", ns),
+				}
+			},
+			test: func(g Gomega, ctx context.Context, k8sClient client.Client, ns string) {
+				dsc := shared.NewDSC("dsc-two", ns)
+				err := k8sClient.Create(ctx, dsc)
+				g.Expect(err).NotTo(Succeed(), "should not allow creation of a second DataScienceCluster")
+			},
+		},
+		{
+			name: "Allows deletion always",
+			setup: func(ns string) []client.Object {
+				return []client.Object{
+					shared.NewDSC("dsc-delete", ns),
+				}
+			},
+			test: func(g Gomega, ctx context.Context, k8sClient client.Client, ns string) {
+				dsc := &dscv1.DataScienceCluster{}
+				key := types.NamespacedName{Name: "dsc-delete", Namespace: ns}
+				g.Expect(k8sClient.Get(ctx, key, dsc)).To(Succeed(), "should find the DataScienceCluster created in setup")
+				g.Expect(dsc.Name).To(Equal("dsc-delete"), "should have the expected name")
+				g.Expect(k8sClient.Delete(ctx, dsc)).To(Succeed(), "should allow deletion of DataScienceCluster")
+			},
+		},
+		{
+			name: "Defaulting: sets ModelRegistry.RegistriesNamespace if empty and Managed",
+			setup: func(ns string) []client.Object {
+				return nil
+			},
+			test: func(g Gomega, ctx context.Context, k8sClient client.Client, ns string) {
+				dsc := shared.NewDSC("dsc-defaulting", ns, WithModelRegistryDefaulting())
+				g.Expect(k8sClient.Create(ctx, dsc)).To(Succeed(), "should allow creation of DataScienceCluster for defaulting test")
+
+				fetched := &dscv1.DataScienceCluster{}
+				g.Eventually(func() string {
+					_ = k8sClient.Get(ctx, types.NamespacedName{Name: "dsc-defaulting", Namespace: ns}, fetched)
+					return fetched.Spec.Components.ModelRegistry.RegistriesNamespace
+				}).Should(Equal(modelregistryctrl.DefaultModelRegistriesNamespace), "should set ModelRegistry.RegistriesNamespace to default when empty and Managed")
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+			t.Logf("Starting test case: %s", tc.name)
+			ctx, env, teardown := shared.SetupEnvAndClient(
+				t,
+				[]envt.RegisterWebhooksFn{
+					datasciencecluster.RegisterWebhooks,
+					dscinitialization.RegisterWebhooks,
+				},
+				30*time.Second,
+			)
+			defer teardown()
```

Comment:
it does not change much, but you can use the `t.Cleanup` helper to register the teardown function to be execute when the tests ends